hadoop-hdfs-project/hadoop-hdfs/src/test/java/io/hops/security/TestUsersGroups.java

```diff
@@ -425,79 +321,124 @@ public void testGetUnixUserGroups() throws Exception {
     HdfsStorageFactory.resetDALInitialized();
     HdfsStorageFactory.setConfiguration(conf);
     HdfsStorageFactory.formatStorage();
+    UsersGroups.createSyncRow();
 
     UserGroupInformation ugi = UserGroupInformation.getLoginUser();
     JniBasedUnixGroupsMappingWithFallback jniGroupsMapping = new JniBasedUnixGroupsMappingWithFallback();
     // Directly get the groups the user belongs to
     List<String> loginUserGroups = jniGroupsMapping.getGroups(ugi.getUserName());
     assertFalse(loginUserGroups.isEmpty());
 
-    UsersGroups.addUserGroupsTx(ugi.getUserName(), new String[]{});
+    UsersGroups.addUserToGroups(ugi.getUserName(), new String[]{});
     List<String> ugiGroups = ugi.getGroups();
+    System.out.println("Count: " + ugiGroups.size() + " Groups: " + Arrays.toString(ugiGroups.toArray()));
     assertFalse(ugiGroups.isEmpty());
     assertThat(ugiGroups, Matchers.equalTo(loginUserGroups));
   }
 
   @Test
-  public void testAddUsers() throws IOException {
+  public void testAddUsers0() throws Exception {
+    testAddUsers(0, 0);
+  }
+
+  @Test
+  public void testAddUsers1() throws Exception {
+    testAddUsers(5, CommonConfigurationKeys.HOPS_UG_CACHE_SIZE_DEFUALT);
+  }
+
+  public void testAddUsers(int cacheTime, int cacheSize) throws IOException {
     Configuration conf = new Configuration();
-    conf.setInt(CommonConfigurationKeys.HOPS_UG_CACHE_SECS, 10);
+    conf.set(CommonConfigurationKeys.HOPS_UG_CACHE_SECS, Integer.toString(cacheTime));
+    conf.set(CommonConfigurationKeys.HOPS_UG_CACHE_SIZE, Integer.toString(cacheSize));
     HdfsStorageFactory.resetDALInitialized();
     HdfsStorageFactory.setConfiguration(conf);
     HdfsStorageFactory.formatStorage();
+    UsersGroups.createSyncRow();
```

Comment:
Isn't supposed to be called in UsersGroups.init?

Reply:
static objects problem